sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -70,7 +70,7 @@ object SimpleAnalyzer extends Analyzer(
     FakeV2SessionCatalog,
     new SessionCatalog(
       new InMemoryCatalog,
-      EmptyFunctionRegistry,
+      FunctionRegistry.builtin,
```

Comment:
Is this fix related?

Reply:
Yes. Without this line of change, we cannot use SimpleAnalyzer to resolve `UnresolvedFunction` because  this is `EmptyFunctionRegistry`.

Reply:
More specifically: we are testing `<` this time: `transform(connectTestRelation.where("id".protoAttr < 0))`, which becomes `UnresolvedFunction(name=<)`

Reply:
Is this guaranteed to be only used in testing?

Reply:
Yes